### PR TITLE
Changes to blockserver to write new sub element to meta.xml

### PR DESCRIPTION
**Pull request by RaiBishal**
### Description of work

Block server to make changes to the XML file after the configuration is edited or new configuration is changed with a new sub-element.

Ticket: https://github.com/ISISComputingGroup/IBEX/issues/4794
### Acceptance criteria

Check if the GUI retains the ticked box after the config is protected even after closing the window

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Has the author taken into account the multi-threaded nature of the code?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?

### Functional Tests

- [ ] Do changes function as described? Add comments below that describe the tests performed.

### Final steps

- [ ] Reviewer has updated the submodule in the main EPICS repo? See **Reviewing work for the subModules of EPICS** in the [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: BlockServer/core/inactive_config_holder.py [hunk 1]
```diff
@@ -98,6 +98,8 @@ def set_config_details(self, details):
                 for args in details["components"]:
                     comp = self.load_configuration(args['name'], True)
                     self.add_component(comp.get_name(), comp)
+            if "isProtected" in details:
+                self._config.meta.isProtected = details["isProtected"]
```
Comment (Tom-Willemsen) [inline BlockServer/core/inactive_config_holder.py]: In the schema, `isProtected` is a non-optional element (`minOccurs="1" maxOccurs="1"`). It should either be optional in the schema (recommended) or not optional here.

File: BlockServer/config/xml_converter.py [hunk 2]
```diff
@@ -419,6 +423,10 @@ def meta_from_xml(root_xml, data):
         if synoptic is not None:
             data.synoptic = synoptic.text if synoptic.text is not None else ""
 
+        isProtected = root_xml.find("./" + TAG_PROTECTED)
+        if isProtected is not None:
+            data.isProtected = isProtected.text if isProtected.text is not None else ""
```
Comment (Tom-Willemsen) [inline BlockServer/config/xml_converter.py]: Is the empty string valid for the `xs:boolean` type? Would be better to explicitly default to `"true"`/`"false"`

File: BlockServer/config/xml_converter.py [hunk 3]
```diff
@@ -167,6 +168,9 @@ def meta_to_xml(data):
             edit = ElementTree.SubElement(edits_xml, TAG_EDIT)
             edit.text = e
 
+        protect_xml = ElementTree.SubElement(root, TAG_PROTECTED)
+        protect_xml.text = repr(data.isProtected).lower()
```
Comment (Tom-Willemsen) [inline BlockServer/config/xml_converter.py]: Why are we using `repr` as opposed to `str`? Most of the time we should be using `str` (or one of the `six` string types if required)

Comment (RaiBishal) [reply BlockServer/config/xml_converter.py]: There were no specific reasons as to why we used `repr` as opposed to `str`.

File: BlockServer/core/inactive_config_holder.py (hunk 1 above)
Comment (RaiBishal) [reply BlockServer/core/inactive_config_holder.py]: Changed `isProtected` to an optional element

File: BlockServer/config/xml_converter.py (hunk 2 above)
Comment (RaiBishal) [reply BlockServer/config/xml_converter.py]: default value set to "false".

**Pull request merged**